source/grid/grid_out.cc

```diff
@@ -3051,18 +3051,22 @@ GridOut::write_vtk(const Triangulation<dim, spacedim> &tria,
   const auto co_faces = relevant_co_faces(tria);
 
   // Write cells preamble
-  const int n_cells = tria.n_active_cells() + faces.size() + co_faces.size();
+  const int n_cells = vtk_flags.draw_cells * tria.n_active_cells() +
+                      vtk_flags.draw_faces * faces.size() +
+                      vtk_flags.draw_co_faces * co_faces.size();
 
   // VTK now expects a number telling the total storage requirement to read all
   // cell connectivity information. The connectivity information is read cell by
   // cell, first specifying how many vertices are required to describe the cell,
   // and then specifying the index of every vertex. This means that for every
   // deal.II object type, we always need n_vertices + 1 integer per cell.
   // Compute the total number here.
-  const int cells_size =
-    tria.n_active_cells() * (GeometryInfo<dim>::vertices_per_cell + 1) +
-    faces.size() * (GeometryInfo<dim>::vertices_per_face + 1) +
-    co_faces.size() * (3); // only in 3d, otherwise it is always zero.
+  const int cells_size = vtk_flags.draw_cells * tria.n_active_cells() *
+                           (GeometryInfo<dim>::vertices_per_cell + 1) +
+                         vtk_flags.draw_faces * faces.size() *
+                           (GeometryInfo<dim>::vertices_per_face + 1) +
+                         vtk_flags.draw_co_faces * co_faces.size() *
+                           (3); // only in 3d, otherwise it is always zero.
 
```

Comment:
Can you add an asserthrow here to make sure that cell_size is non zero? With the flags, one could in principle have no output at all.

Reply:
@luca-heltai Thanks for the review! I have added the assert and a test, which I I think matches your description.

Reply:
I think one might also want to have an assertion somewhere that not all three of the flags are `false` at the same time.